src/layout/Awaiter.js

```diff
@@ -0,0 +1,10 @@
+import React from 'react'
+import { observer } from 'mobx-react'
+
+export default observer(({ promise, children }) =>
+  promise.case({
+    pending: () => <div>Loading...</div>,
+    rejected: error => <div>Ooops.. {error}</div>,
+    fulfilled: value => <div>{children(value)}</div>,
```

Comment:
I'm afraid that having mixed children as functions and as children of nodes and as children of components can be confusing.

Reply:
Render props are the new hotness  😄 
https://reactjs.org/docs/render-props.html#using-props-other-than-render
https://medium.com/dailyjs/reacts-%EF%B8%8F-new-context-api-70c9fe01596b